src/components/VideoPlayer/VideoPlayer.tsx

```diff
@@ -0,0 +1,81 @@
+import * as React from "react";
+import bem from "../../utils/bem";
+import { VideoPlayerAspectRatios, VideoPlayerTypes } from "./VideoPlayerTypes";
+
+export interface VideoPlayerProps {
+  /** Optional aspect ratio prop to control the sizing of the video player; if omitted, the video player defaults to `sixteen-by-nine` */
+  aspectRatio?: VideoPlayerAspectRatios;
+  /** Optional additional attributes passed to the video player iframe */
+  attributes?: { [key: string]: any };
+  /** Optional className you can add in addition to `video-player` */
+  className?: string;
+  /** Optional title to be added to the `<iframe>` element for improved accessibility; this title should describe in a few words the content of the video; if omitted, a generic title will be added */
+  iframeTitle?: string;
+  /** Modifiers array for use with BEM. See how to work with modifiers and BEM here: http://getbem.com/introduction/ */
+  modifiers?: string[];
+  /** Required YouTube or Vimeo video ID. This value can be pulled from a video's YouTube or Vimeo URL. */
+  videoId: string;
+  /** Required. Used to specify which video service is being used. */
+  videoType: VideoPlayerTypes;
+}
+
+export default function VideoPlayer(
+  props: React.PropsWithChildren<VideoPlayerProps>
+) {
+  const {
+    aspectRatio,
+    attributes = {},
+    className,
+    iframeTitle,
+    modifiers = [],
+    videoId,
+    videoType,
+  } = props;
+
+  let embedCode;
+  let iframeTitleFinal;
+  let videoSrc;
+
+  {
+    aspectRatio && modifiers.push(aspectRatio);
+  }
+
+  switch (videoType) {
```

Comment:
@bigfishdesign13 Wouldn't it be better to just have the consuming app pass the video src url, and then we wouldn't need to hardcode the links to vimeo and youtube?

Reply:
@nypl-wluisi If the direct URL for the video on YouTube or Vimeo was passed, how would you handle the embed code?  The structure of the video viewing page and the video embed page are different.

View: `https://www.youtube.com/watch?v=0fpNksg7-2I`
Embed: `https://www.youtube.com/embed/0fpNksg7-2I`

Reply:
You could try to parse out the video id from the specific video type URL. But, I think it makes sense to pass in just the ID because there are specific parameters that will make video embed renderings consistent throughout different applications.